get_github_coderabbit_review/coderabbit_fetcher/cli/main.py

```
@@ -1,121 +1,596 @@
+    def setup_github_client(self) -> None:
+        """Initialize and validate GitHub client."""
+        try:
+            self.github_client = GitHubClient()
+            logger.info("GitHub CLI authentication verified")
+        except GitHubAuthenticationError as e:
+            print(f"❌ GitHub Authentication Error: {e}", file=sys.stderr)
+            print("\n💡 To fix this issue:", file=sys.stderr)
+            print("   1. Install GitHub CLI: https://cli.github.com/", file=sys.stderr)
+            print("   2. Run: gh auth login", file=sys.stderr)
+            print("   3. Follow the authentication prompts", file=sys.stderr)
+            raise CLIError("GitHub CLI authentication required")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)